docs/reference/injector.md

```diff
@@ -0,0 +1,173 @@
+  namespace: default-enabled
+spec:
+  template:
+    spec:
+      containers:
+      - name: default-behavior
+        image: tutum/curl
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: override-default-enabled
+  namespace: default-enabled
+spec:
+  template:
+    metadata:
+      annotations:
+        appmesh.k8s.aws/sidecarInjectorWebhook: disabled // this will override the default and disable inject sidecar
+    spec:
+      containers:
+      - name: override-default-enabled
+        image: tutum/curl
+```
+
+`appmesh.k8s.aws/sidecarInjectorWebhook: disabled`: The sidecar injector will not inject the sidecar into pods by default. Add the `appmesh.k8s.aws/sidecarInjectorWebhook` annotation with value `enabled` to the pod template spec to override the default and enable injection.
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: default-disabled
+  labels:
+    mesh: default-disabled
+    appmesh.k8s.aws/sidecarInjectorWebhook: disabled
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: default-behavior // sidecar will be not injected as namespace has sidecar injection disabled
+  namespace: default-enabled
+spec:
+  template:
+    spec:
+      containers:
+      - name: default-behavior
+        image: tutum/curl
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: override-default-disabled
+  namespace: default-disabled
+spec:
+  template:
+    metadata:
+      annotations:
+        appmesh.k8s.aws/sidecarInjectorWebhook: enabled // this will override the default and inject sidecar
+    spec:
+      containers:
+      - name: override-default-disabled
+        image: tutum/curl
+```
+
+
+## Envoy injection for virtual gateways
+
+AWS App Mesh supports virtual gateway resource to allow resources that are outside of your mesh to communicate to resources that are inside of your mesh. The virtual gateway represents an Envoy proxy running in the Kubernetes cluster. Unlike a virtual node, which represents Envoy running with an application, a virtual gateway represents Envoy deployed by itself. App Mesh Kubernetes controller supports injecting Envoy and virtual gateway configuration.
+
+App Mesh Kubernetes controller uses podSelector to designate Virtual Gateway membership. If you create an pod with labels matching the pod selector labels in a virtual gateway spec, the controller will inject the Envoy configuration to the pod/envoy container and override the default container image by default.
+
+Also, since a pod may contain multiple containers, the controller relies on the container name `envoy` to determine, which container to mutate for virtual gateway configuration.
+
+To use the controller to inject virtual gateway configuration, add podSelector to your virtual gateway and set the container name to `envoy`:
+
+```
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualGateway
+metadata:
+  name: ingress-gw
+  namespace: ns
+spec:
+  namespaceSelector:
+    matchLabels:
+      gateway: ingress-gw
```

Comment:
We're missing this in above VirtualGateway description (i.e.,) the need to have a namespace label for VG.

Reply:
Done, added to the description